src/with-defaults.ts

```diff
@@ -5,17 +5,34 @@ import type {
   RequestParameters,
 } from "./types.js";
 import { graphql } from "./graphql.js";
+import type { DocumentTypeDecoration } from "@graphql-typed-document-node/core";
 
 export function withDefaults(
   request: typeof Request,
   newDefaults: RequestParameters,
 ): ApiInterface {
   const newRequest = request.defaults(newDefaults);
   const newApi = <ResponseData>(
-    query: Query | RequestParameters,
+    query:
+      | Query
+      | (String & DocumentTypeDecoration<unknown, unknown>)
+      | RequestParameters,
     options?: RequestParameters,
   ) => {
-    return graphql<ResponseData>(newRequest, query, options);
+    const innerQuery =
+      typeof query === "string"
+        ? query
+        : // Allows casting String & DocumentTypeDecoration<unknown, unknown> to
+          // string. This could be replaced with an instanceof check if we had
+          // access to a shared TypedDocumentString. Alternatively, we could use
+          // string & TypedDocumentDecoration<unknown, unknown> as the external
+          // interface, and push `.toString()` onto the caller, which might not
+          // be the worst idea.
+          String.prototype.isPrototypeOf(query)
+          ? query.toString()
+          : (query as RequestParameters);
```

Comment:
Left some notes on this at: https://github.com/octokit/graphql.js/pull/609#:~:text=in%20similar%20projects.-,Open%20remarks,-I%20am%20not